sklearn/tests/test_naive_bayes.py

```diff
@@ -438,6 +445,61 @@ def test_discretenb_coef_intercept_shape(cls):
     assert clf.intercept_.shape == (1,)
 
 
+@pytest.mark.parametrize("cls", [MultinomialNB, ComplementNB, BernoulliNB,
+                                 CategoricalNB])
+def test_discretenb_degenerate_one_class_case(cls):
+    # The first axis of most array attributes of a discrete naive Bayes
+    #     classifier should have one element per class. Exceptions include:
+    #     ComplementNB.feature_all_, CategoricalNB.n_categories_.
+    # Confirm that this is the case for binary problems and degenerate
+    #     one-class problems when fitting with `fit` or `partial_fit`.
+    # Non-regression test for handling degenerate one-class case.
+
+    attr_list = ['classes_', 'class_count_', 'class_log_prior_',
+                 'feature_count_', 'feature_log_prob_']
+    X = [[1, 0, 0], [0, 1, 0], [0, 0, 1]]
+    y = [1, 1, 2]
+
+    def test_first_axis_lengths(expected_first_axis_length):
```

Comment:
if needed you can put the test outside as a private check, and use the parametrization